### PR TITLE
Charlesthebird/pkce auth

### PR DESCRIPTION
**Pull request by Charlesthebird**
This addresses the dev-portal-starter part of https://github.com/solo-io/gloo-mesh-enterprise/issues/9807
The changes to your Keycloak client settings to enable PKCE authorization code flow include:
- Setting `Access Type` to `public`.
- Setting `Standard Flow Enabled` to true.
- Setting `Direct Access Grants Enabled` to true.
- Setting `Valid Redirect URIs` to `*`.
- Setting `Proof Key for Code Exchange Code Challenge Method=S256` under `Advanced Settings`

It also updates the environment variables so that the `end_session_endpoint` is used to fully logout.

Video showing login flow working:

https://github.com/solo-io/dev-portal-starter/assets/4720646/b6d75356-ddf8-43b4-8631-0e058825f475

### CODE REVIEW

File: projects/ui/src/Components/Structure/HeaderSectionLoggedIn.tsx [hunk 1]
```diff
@@ -1,23 +1,31 @@
...
   );
 
+  const [searchParams, setSearchParams] = useSearchParams();
+  useEffect(() => {
+    const newSearchParams = new URLSearchParams(searchParams);
+    newSearchParams.delete("state");
+    newSearchParams.delete("session_state");
+    newSearchParams.delete("code");
+    setSearchParams(newSearchParams);
+  }, [setSearchParams]);
```
Comment (Worfle) [inline projects/ui/src/Components/Structure/HeaderSectionLoggedIn.tsx:27]: Is there a reason setSearchParams would be changing, or is this only to do on component load after the setSearchParam is settled?

Honestly, this part is nicely terse, but there's no context nearby for what's happening and its a bit hard to follow for why we are doing this/what the goal of this piece of code is.

**Review by Worfle — APPROVED** [review]
There's a lot of security content, including some explanations.   I didn't see anything that seemed to give away too much info, but I guess i'm wary of the UI taking on any real security work.

Again though, I didn't see anything and functionally it almost all seems good to me.

**Review by williamgrh — APPROVED** [review]

**Review by toreysoloio — APPROVED** [review]

File: projects/ui/src/Components/Structure/HeaderSectionLoggedIn.tsx (hunk 1 above)
Comment (Charlesthebird) [reply projects/ui/src/Components/Structure/HeaderSectionLoggedIn.tsx:27]: `setSearchParams` is just in the dependency array for completeness, it shouldn't change.

I didn't add any explanations here, but this basically cleans up the URL after it is redirected from the login screen since it has all that info in it.
